### DISCUSSION TITLE
Not able to use Jinja template in PostgresOperator Airflow in SQL file naming

### DISCUSSION THREAD

**Discussion by kameshkotwani** [apache/airflow#27074] [Q&A]
Have been searching for this issue, but all the answers are old and not relevant to my issue.
I am trying to pass a ```sql``` parameter as a Jinja Template, but not able to. I am a bit new to Airflow, here is the code, I am not sure if the Jinja templating works in PostgresOperator or not. If there is any other way we can pass please let me know, again I am not trying to pass the jinja template inside the SQL query or file, I am passing the name of the SQL file using the Jinja template which it is not able to render.

PostgerOperatorCode:
```python
write_to_postgres = PostgresOperator(
        task_id ="write_to_postgres",
        postgres_conn_id="mypostgres",
        #TODO investigate, why this is not working
        sql ="queries{{ execution_date.hour }}{{ execution_date.day }}.sql"
    ) 
```
The templating is working on BashOperator, and PythonOpertor, but not in Postgres.
Airflow Webserver in UI says, not able to render templates.

[![Postgres render ][1]][1]

BashOperatorCode:
```bash
 extract_gz = BashOperator(
        task_id="extract_downloaded_gz",
        bash_command="gunzip --force /opt/airflow/dags/wikipageviews{{ execution_date.hour }}{{ execution_date.day }}.gz"
    )
```

For BashOperator for the same code it works:
[![BashOperator][2]][2]

  [1]: https://i.stack.imgur.com/BSwOc.png
  [2]: https://i.stack.imgur.com/8g77n.png

**Comment by potiuk** [comment]
It's currently not possible to use JINJA in file name. The sql parameter works in the way that when your "sql" parameter ends with `.sql`, it will try to load it from a file and it does not try to pass the name via JINJA templating.

This is done here: https://github.com/apache/airflow/blob/main/airflow/models/abstractoperator.py#L171

You might want to attempt to change it - it should be as easy as checking if there is a "{{" in the parameter and passing it through JINJA - if you want to add it as a feature, you might want to modify your local version of airflow to test if it works and submit it as PR - this would be a nice contribution. It is slightly incompatible (if someone DOES have a file with {{}} it might create a problem, but I think this problem could be ignored (when you create a PR with it it might be discussed then and possibly a flag enabling it might be added if others think thi might be an issue)
